vulkaninfo/vulkaninfo.cpp

```diff
@@ -853,10 +864,29 @@ int main(int argc, char **argv) {
     }
 #endif
 
+    int return_code = 0;  // set in case of error
     uint32_t selected_gpu = 0;
     bool show_formats = false;
     char *output_path = nullptr;
 
+    // Figure out the name of the executable, pull out the name if given a path
+    // Default is `vulkaninfo`
+    std::string executable_name = "vulkaninfo";
+    if (argc >= 1) {
+#ifdef _WIN32
+        const char *platform_separator = "\\";
+#else
+        const char *platform_separator = "/";
+#endif
+        executable_name = std::string(argv[0]);
+        // don't include path separator
+        auto last_occurrence = executable_name.rfind(platform_separator);
+        if (last_occurrence != std::string::npos && last_occurrence < executable_name.length()) {
+            auto sub_length = executable_name.length() - last_occurrence - 1;
+            executable_name = executable_name.substr(last_occurrence + 1, sub_length);
```

Comment:
Is the calculation of `sub_length` necessary?  I'd think `executable_name.substr(last_occurrence + 1)` would be sufficient.

Reply:
Yep, turns out to be necessary.